stl/inc/xutility

```diff
@@ -4667,6 +4667,63 @@ _NODISCARD bool lexicographical_compare(
 }
 #endif // _HAS_CXX17
 
+#if defined(__cpp_impl_three_way_comparison) && defined(__cpp_lib_concepts)
+// FUNCTION TEMPLATE lexicographical_compare_three_way
+template <class _InIt1, class _InIt2, class _Cmp>
+constexpr auto lexicographical_compare_three_way(_InIt1 _First1, _InIt1 _Last1, _InIt2 _First2, _InIt2 _Last2,
+    _Cmp _Comp) -> decltype(_Comp(*_First1, *_First2)) {
+    _Adl_verify_range(_First1, _Last1);
+    _Adl_verify_range(_First2, _Last2);
+    auto _UFirst1      = _Get_unwrapped(_First1);
+    const auto _ULast1 = _Get_unwrapped(_Last1);
+    auto _UFirst2      = _Get_unwrapped(_First2);
+    const auto _ULast2 = _Get_unwrapped(_Last2);
+    using _UIt1        = decltype(_UFirst1);
+    using _UIt2        = decltype(_UFirst2);
+
+    if constexpr (conjunction_v<is_same<_Cmp, compare_three_way>, is_pointer<_UIt1>, is_pointer<_UIt2>,
+                      disjunction<
+                          conjunction<is_same<remove_pointer_t<_UIt1>, byte>, is_same<remove_pointer_t<_UIt2>, byte>>,
+                          conjunction<_Is_character<remove_pointer_t<_UIt1>>, is_unsigned<remove_pointer_t<_UIt1>>,
+                              _Is_character<remove_pointer_t<_UIt2>>, is_unsigned<remove_pointer_t<_UIt2>>>>>) {

```

Comment:
`_Is_character` is also sensitive to cv-qualifiers; `_Is_character<const unsigned char>` is `false`:

https://github.com/microsoft/STL/blob/d4ee5c3fb4abfe8492f2108384d6f4e0fd0afeae/stl/inc/xutility#L4106-L4121

Note that this is an interface difference between this internal type trait and the primary type categories (e.g. `is_integral` is specified to ignore top-level cv-qualifiers).

Perhaps you should introduce helper typedefs for `remove_const_t<remove_pointer_t<_UIt1>>` and similarly for `_UIt2`. (If the iterators aren't pointers after unwrapping, these won't be element types, but you simply won't inspect them. Alternatively, I suppose you could use iterator traits machinery to get the value types.)

Reply:
Fixed